### DISCUSSION TITLE
How to add new table to zumo-quickstart TodoItem Xamarin.forms

### DISCUSSION THREAD

**Discussion by ghost** [Azure/azure-mobile-apps#291] [Q&A] [answered]
In order to learn how to add tables to Azure SQL Db and connect to them via MobileServiceClient , .GetTable and InsertAsync I am testing with the ZumoQuickstart xamarin.forms sample. I have completed the ZumoQuickstart tutorial with no problems. I add my own table either using the script from the TodoItem table and changing the table names or in SSMS. Replicate the relevant code in TodoService but with my table name and try an InsertAsync. It fails with error from LoggingHandler of "Newtonsoft.Json.JsonReaderException: 'Unexpected character encountered while parsing value: <. Path '', line 0, position 0.'" 
or with exception at InsertAsync line of  "Microsoft.WindowsAzure.MobileServices.MobileServiceInvalidOperationException: '<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /tables/SurveyResponse</pre>
</body>
</html>

' or  via postman GET with 404 Not Found."

There are no table controllers in the ZumoQuickstart, is that done by "mTable = mClient.GetTable<TodoItem>( );"? Is there something else I need to do in the App Service or the mobile Client app to "register a new table" or could there be some kind of mmismatch between the Azure table and the mobile client class?

**Comment by ghost** [comment, marked as answer]
Please don't bother answering above question. I am building the complete App from Adrians book and will merge my Xamarin.forms App into it. This I hope will improve my knowledge and skill (but no doubt I'll still post more dumb questions!)

**Reply by adrianhall** [reply]
Just to clarify the error (and then I'll close the discussion), the error you are seeing is because there isn't a table controller.  You have two sides of the client-server protocol.  On the server side, you have an OData endpoint called `/tables/{tableName}` - this is normally an ASP.NET controller, but can also be a NodeJS endpoint, depending on which language you are using service side.  On the client side, you have a `mClient.GetTable<T>()` call.  This does not define the table.  It just accesses the table defined on the server side.

... and I am working on a revamp of the book in parallel to the codebase, so that when the new codebase goes GA, I'll be able to release the book as well.

**Reply by ghost** [reply]
Hi Adrian. Thanks for the clarification as I started working through your book it became apparent. Looking forward to a revamped book!
